src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs

```diff
@@ -168,14 +171,28 @@ public static IFhirServerBuilder AddSqlServer(this IFhirServerBuilder fhirServer
                 .AsSelf();
 
             services.Add<DefragWatchdog>().Singleton().AsSelf();
-
             services.Add<CleanupEventLogWatchdog>().Singleton().AsSelf();
-
             services.Add<TransactionWatchdog>().Scoped().AsSelf();
             services.AddFactory<IScoped<TransactionWatchdog>>();
-
             services.Add<InvisibleHistoryCleanupWatchdog>().Singleton().AsSelf();
 
+            // Conditionally register GeoReplicationLagWatchdog based on core feature flag
+            services.AddSingleton<GeoReplicationLagWatchdog>(provider =>
+            {
+                var coreFeatures = provider.GetService<IOptions<CoreFeatureConfiguration>>();
+
+                if (coreFeatures?.Value?.EnableGeoRedundancy == true)
+                {
+                    var sqlRetryService = provider.GetRequiredService<ISqlRetryService>();
+                    var logger = provider.GetRequiredService<ILogger<GeoReplicationLagWatchdog>>();
+                    var mediator = provider.GetRequiredService<IMediator>();
+                    var schemaInformation = provider.GetRequiredService<SchemaInformation>();
+                    return new GeoReplicationLagWatchdog(sqlRetryService, logger, mediator, schemaInformation);
+                }
+
+                return null;
+            });
```

Comment:
I'm still wondering if it would just be better (and easier) to always fully register the component and only resolve it when we need it (config is on)

Reply:
Changed to fully registered to keep it simple